src/components/EditProfile.js

```diff
@@ -0,0 +1,233 @@
+/* eslint-disable no-unused-vars */
+/* eslint-disable react/prop-types */
+/* eslint-disable no-undef */
+import React, {useState, useEffect} from "react";
+import {useNavigate} from "react-router-dom";
+import {getStorage, ref, uploadBytes, getDownloadURL} from "firebase/storage";
+import {
+  getDoc,
+  doc,
+  updateDoc,
+} from "firebase/firestore";
+
+import {
+  EditWrapper,
+  EditNavAllWrapper,
+  EditNavLeftWrapper,
+  EditNavRightWrapper,
+  BackButton,
+  EditTitle,
+  CancelButton,
+  SaveButton,
+  InputTextWrapper,
+  InputDescTextWrapper,
+  InputTextTitle,
+  InputText,
+  InputFileWrapper,
+  InputFileTitle,
+  LabelFile,
+  PhotoDisplay,
+  InputFile,
+  InputDesc,
+} from "../styles/EditProfile.module";
+
+function EditProfile(props) {
+  const [userData, setUserData] = useState("");
+  const [displayPhoto, setDisplayPhoto] = useState("");
+  const [selectedFile, setSelectedFile] = useState("");
+  const [newName, setNewName] = useState("");
+  const [newDescription, setNewDescription] = useState("");
+  const [newLink, setNewLink] = useState("");
+  const [newUserPhotoName, setNewUserPhotoName] = useState("");
+
+  const storage = getStorage(props.app);
+  const redirect = useNavigate();
+
+  const getUserData = (userId) => {
+    // eslint-disable-next-line no-unused-vars
+    // change from onSnapShot to getDoc to prevent infinite loop
+    const unsub = getDoc(doc(props.db, "user", userId), (doc) => {
+      if (!props.uid) {
+        return;
+      } else if (props.uid && !userData) {
```

Comment:
I guess you use props.uid to check if a user is logged in. Then this is weird to check here. It is supposed to be checked on high-level components like App.js.
And you have already passed 'userId' in this function. You need to standardize your style for 'getXXXX' function in other component. In my opinion, checking props.uid in App.js first, then you could just use uid from props instead of passing as a parameter.

Reply:
You need to manage auth in router if the user should not be accessed in the component.